db/seeds.rb

```diff
@@ -37,6 +37,11 @@ def audit_modified?(type, name, attributes = {})
 
 def create_filters(role, collection)
   collection.group_by(&:resource_type).each do |resource, permissions|
+    # check if the filter exists already
+    next if Filter.joins(:permissions).find_by("permissions.id" => permissions.map(&:id),
+                                               :role_id => role.id
```

Comment:
That's a good point, the fact that it's a seed helper suggests idempotency. But in this case I think you should check all attributes since we don't have uniq identifier as for roles. The method would have to accept also search and taxonomies. Or at least check that these attributes were note modified, basically that the filter is unlimited. There's already an [unlimited scope](https://github.com/theforeman/foreman/blob/develop/app/models/filter.rb#L47) that could be used.